packages/synapse-client/src/util/waitForAsyncResult.ts

```diff
@@ -30,5 +31,20 @@ export async function waitForAsyncResult<
     // Exponential backoff for the next request
     currentDelayMs = Math.min(currentDelayMs * 2, maxDelayMs)
   }
+  if ('jobState' in response && response.jobState === 'FAILED') {
```

Comment:
This only happens if this utility is used to call [`GET /asynchronous/job/{jobId}`](https://rest-docs.synapse.org/rest/GET/asynchronous/job/jobId.html). When this happens, throw an error.